main.go

```diff
@@ -91,26 +102,56 @@ func main() {
 			tlsConn, err := vhost.TLS(c)
 			if err != nil {
 				log.Printf("Error accepting new connection - %v", err)
+				return
 			}
+			defer tlsConn.Close()
```

Comment:
In most cases when writing low level tcp listener I move serving login to separate function which both prevents bugs with scopes and is more clean.

```go
for {
   c, err := ln.Accept()
  // ...
  go serve(c)
}

func serve(c net.Conn) {}
```

Reply:
Make sense, moved it into `serveTLS` function.